### ISSUE TITLE
module 'tensorflow.python.compiler.tensorrt.utils' has no attribute 'versionTupleToString'

### ISSUE THREAD

**Issue by DocGarbanzo** [tensorflow/tensorflow#53396]
# Tensorrt conversion fails on Ubuntu 

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): 20.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): 2.7 binary (via pip)
- TensorFlow version (use command below): 2.7
- Python version: v2.7.0-rc1-69-gc256c071bb2 2.7.0
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 11.5
- GPU model and memory:  Geforce RTX 2060 Super

**Describe the current behavior**
The tensorrt conversion fails when the installed version 7.2.3 is higher than the linked version 7.2.2, because the python code in the logger statements is wrong.

**Describe the expected behavior**
Tensorrt conversion is expected to succeed.

**[Contributing](https://www.tensorflow.org/community/contribute)**

- Do you want to contribute a PR? (yes/no): no
- Briefly describe your candidate solution(if contributing):

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

Not needed here, just look at `trt_convert.py:229ff` - the function `trt_utils.versionTupleToString` is called multiple times but is not defined, there is only `trt_utils._version_tuple_to_string`. The logger only gets called when there is a version mismatch.  A simple 

```python
versionTupleToString = _version_tuple_to_string
```
in `utils.py` will solve it.

File `trt_convert.py:229ff`:
```python
  def raise_trt_version_deprecated(version_type, trt_version):
    assert version_type in [
        "linked", "loaded"
    ], ("Incorrect value received for version_type: %s. Accepted: ['linked', "
        "'loaded']") % version_type

    logging.error(
        "The {version_type} version of TensorRT: `{trt_version}` has now "
        "been removed. Please upgrade to TensorRT 7 or more recent.".format(
            version_type=version_type,
            trt_version=trt_utils.versionTupleToString(trt_version)))

    raise RuntimeError("Incompatible %s TensorRT versions" % version_type)

  if not trt_utils.is_linked_tensorrt_version_greater_equal(7, 0, 0):
    raise_trt_version_deprecated("linked", linked_version)

  if not trt_utils.is_loaded_tensorrt_version_greater_equal(7, 0, 0):
    raise_trt_version_deprecated("loaded", loaded_version)

  if (loaded_version[0] != linked_version[0] or
      not trt_utils.is_loaded_tensorrt_version_greater_equal(*linked_version)):
    logging.error(
        "Loaded TensorRT %s but linked TensorFlow against TensorRT %s. A few "
        "requirements must be met:\n"
        "\t-It is required to use the same major version of TensorRT during "
        "compilation and runtime.\n"
        "\t-TensorRT does not support forward compatibility. The loaded "
        "version has to be equal or more recent than the linked version.",
        trt_utils.versionTupleToString(loaded_version),
        trt_utils.versionTupleToString(linked_version))
    raise RuntimeError("Incompatible TensorRT major version")

  elif loaded_version != linked_version:
    logging.info(
        "Loaded TensorRT %s and linked TensorFlow against TensorRT %s. This is "
        "supported because TensorRT minor/patch upgrades are backward "
        "compatible.", trt_utils.versionTupleToString(loaded_version),
        trt_utils.versionTupleToString(linked_version))

```

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by sachinprasadhs** [comment]
Did you try with `Tensorrt` version 7.2.2, also check the compatibility and limitations of Tensorrt 7.2.2 [here](https://docs.nvidia.com/deeplearning/tensorrt/release-notes/tensorrt-7.html#rel_7-2-2).

**Comment by DocGarbanzo** [comment]
@sachinprasadhs - I didn't check w/t 7.2.2, but if you look at the code above, you see that the function `trt_utils.versionTupleToString` is only being called if the runtime version is different than the linked version. The problem is, that this function doesn't exist, adding it to `utils.py` will solve the issue. The code says, it should be working with the same major version and any equal or larger minor version in the runtime. Which it did for me when I made the quick fix described above.

**Comment by DocGarbanzo** [comment]
@sachinprasadhs, @sanjoy  - do you have any update on this? Could you please acknowledge that this is a bug and will get fixed. It's basically a blocker when you want to upgrade to tf 2.7 and you support tensorrt.

**Comment by WingsOfPanda** [comment]
is this fixed in tf2.8?

**Comment by DocGarbanzo** [comment]
@sachinprasadhs, @sanjoy  - do you have any update on this?

**Comment by apivovarov** [comment]
it was fixed in r2.8 and 2.8.0
- `utils._version_tuple_to_string` was renamed to `utils.version_tuple_to_string`
- `trt_convert.py` uses `trt_utils.version_tuple_to_string` instead of `trt_utils.versionTupleToString`.

**Comment by sachinprasadhs** [comment]
As mentioned by @apivovarov , It was. fixed in TF 2.8, you can find the source code below.
https://github.com/tensorflow/tensorflow/blob/v2.8.0/tensorflow/python/compiler/tensorrt/trt_convert.py#L261-L262
Here is the fix which resolved the issue. https://github.com/tensorflow/tensorflow/commit/b523f96d480fd6783aa16c1ce55419a6e4cd9cc6

**Comment by sachinprasadhs** [comment]
Cherrypicks will be usually done when there is any security vulnerability, in this case you can use Tensorflow 2.8, since the fix is already available and Tensorflow 2.9 release is around the corner as well.

**Comment by DocGarbanzo** [comment]
Thanks @sachinprasadhs!
